vault/logical_system.go

```diff
@@ -1309,6 +1317,19 @@ func (b *SystemBackend) handleMountTuneWrite(ctx context.Context, req *logical.R
 	return b.handleTuneWriteCommon(ctx, path, data)
 }
 
+func augmentTuneWriteParams(paramIn interface{}) ([]string, error) {
+	newVal := paramIn.([]string)
+	var outputSlice []string
+	for _, v := range newVal {
```

Comment:
Is this the behaviour we want?  My thinking was that we should only expand commas if we were given a single string input, but I haven't looked to see what we do elsewhere.

Reply:
Based on what I have understood (please correct me if I am wrong), the schema for these values is TypeCommaStringSlice. Then, for example, the following two commands should work:
1 -   "vault auth tune -audit-non-hmac-request-keys=value1 -audit-non-hmac-request-keys=value2 -audit-non-hmac-request-keys=value3 userpass" 
2-  "vault auth tune -audit-non-hmac-request-keys=value1 -audit-non-hmac-request-keys=value2,value3 userpass"

and the result for the above two commands needs to match.
"audit_non_hmac_request_keys": [
        "value1",
        "value2",
        "value3"
      ],

Reply:
Yeah, I suspect your intuition was correct here.  I think I was just feeling like there should be some way to provide a value that contains a comma if needed, but upon further reflection I can't see why we'd need to.